subsys/fs/nvs/nvs.c

```diff
@@ -88,7 +88,7 @@ static void nvs_lookup_cache_invalidate(struct nvs_fs *fs, uint32_t sector)
 /* nvs_al_size returns size aligned to fs->write_block_size */
 static inline size_t nvs_al_size(struct nvs_fs *fs, size_t len)
 {
-	uint8_t write_block_size = fs->flash_parameters->write_block_size;
+	size_t write_block_size = fs->flash_parameters->write_block_size;
```

Comment:
If there is concern that `write_block_size`, which has `size_t` type in the `struct flash_parameters` definition, can overflow the local definition, which is suitable to hold the design allowed values (https://github.com/zephyrproject-rtos/zephyr/pull/77569#issuecomment-2312766152), then `if` should be placed here, checking the value and reporting error; but I think that proper place for the warning would be in `nvs_mount`, not here.

Reply:
The goal is to avoid the local variable overflow for all cases.

Reply:
The original code is going to get flagged by the static analysis due to the size mismatch of this assignment.

Reply:
> The original code is going to get flagged by the static analysis due to the size mismatch of this assignment.

That can be fixed with cast, but will not change design constrains.

> The goal is to avoid the local variable overflow for all cases.

Look, I do not have any solid arguments against using `size_t` here, so I guess the change itself is ok; but the fact that this can overflow comes from misuse where improper value got here without being vatted against the design constrains.

Reply:
understood. But lets move that discussion to the root you reference. 

We have raised an RFC #77576 for that discussion.